voyager_archive/templates/voyager_archive/invoice_display.html

```diff
@@ -0,0 +1,131 @@
+            <tr>
+                <td class="label">Created</td>
+                <td>{{ inv_header.create_date }} by {{ inv_header.create_operator_id }}</td>
+            </tr>
+            <tr>
+                <td class="label">Modified</td>
+                <td>{{ inv_header.modify_date }} by {{ inv_header.modify_operator_id }}</td>
+            </tr>
+        </table>
+    </div>
+    <div class="column">
+        <table class="data-display">
+            <tr>
+                <td class="label">Original Currency</td>
+                <td>
+                    {% if inv_header.currency_code == "USD" %}
+                    USD
+                    {% else %}
+                    {{ inv_header.currency_code }} (1 USD = {{ inv_header.conversion_rate|floatformat:2 }} {{ inv_header.currency_code }})
+                    {% endif %}
+                </td>
+            </tr>
+            <tr>
+                <td class="label">Line Item Subtotal</td>
+                <td>USD ${{ inv_header.line_item_subtotal|floatformat:2 }}</td>
+            </tr>
+            <tr>
+                <td class="label">Adjustments Subtotal</td>
+                <td>USD ${{ inv_header.adjustments_subtotal|floatformat:2 }}</td>
+            </tr>
+            <tr>
+                <td class="label">Invoice Total</td>
+                <td>USD ${{ inv_header.invoice_total|floatformat:2 }}</td>
+            </tr>
+            <tr>
+                <td class="label">-----</td>
+                <td>-----</td>
+            </tr>
+            <tr>
+                <td class="label">Line Items</td>
+                <td>{{ inv_header.line_item_count }}</td>
+            </tr>
+        </table>        
+    </div>
+</div>
+
+{% if inv_adjustments %}
+<div class="row">
+    <div class="column">
+        <h2>Invoice Adjustments</h2>
+        <table class="data-display">
+            <tr>
+                <th>Description</th>
+                <th>Ledger & Fund</th>
+                <th>Amount</th>
+            </tr>
+            {% for adjustment in inv_adjustments %}
+            <tr>
+                <td>{{ adjustment.description }}</td>
+                <td>{{ adjustment.ledger_name }} / {{ adjustment.fund_name }}</td>
+                <td>USD ${{ adjustment.usd_amount|floatformat:2 }}</td>
+            </tr>
+        {% endfor %}
+        </table>
+    </div>
+</div>
+{% endif %}
+
+{% if inv_lines %}
+<div class="row">
+    <div class="column">
+        <h2>Invoice Lines</h2>
+        <table class="data-display">
+            <tr>
+                <th>Status</th>
+                <th>Title</th>
+                <th>Piece Identifier</th>
+                <th>Ledger & Fund</th>
+                <th>Amount</th>
+            </tr>
+        {% for line in inv_lines %}
+            <tr>
+                <td>{{ line.status }}</td>
+                <td>{{ line.bib_id }} {{ line.title }}</td>
+                <td>{% if line.piece_identifer %}{{ line.piece_identifier }}{% endif %}</td>
```

Comment:
Another question: what is this if statement doing? I noticed that if I take it out, two-letter codes (TM, TS) are visible in the Invoice Lines table.

Reply:
@ztucker4 This is intended to be conditional display, shorthand for `if line.piece_identifier is not None`.  String values like this from the `InvoiceLineView` object (and others) will be `None` if not populated, and the template will display "None", which I don't want.

However... you're right, this is not behaving as expected per [the documentation](https://docs.djangoproject.com/en/4.1/ref/templates/builtins/#if).  There is a value but the `if` test is not recognizing it.  Bizarrely, if I change the test to `{% if line.piece_identifer is None %}`... the value **displays**!

For now, since I'm obviously not understanding something, I've removed the test.  _*Excellent question!*_